zerver/actions/streams.py

```diff
@@ -69,6 +70,11 @@
 def do_deactivate_stream(
     stream: Stream, log: bool = True, *, acting_user: Optional[UserProfile]
 ) -> None:
+    # Error if this stream is used for reporting messages. Note that this also works for
+    # merge_streams admin command on the command-line.
+    if Realm.objects.filter(id=stream.realm_id, reportmsg_stream=stream).exists():
+        raise JsonableError(_("Cannot deactivate abuse report stream."))
+
```

Comment:
Do we need one of these for the signups and the notifications streams, too?

Reply:
"above my paygrade" LoL - you and core team tell me... IMHO yes they should all error - surprises are no fun.

I'd politely request we not add more overhead to this PR, and instead create a new issue, which I'm happy to do next (and quickly). Generally, with 3 examples, it's probably time to elevate/refactor these kinds of streams to a new subclass with special properties (no archival to start, but perhaps unified settings mgmt?)...